File: frontend/src/api/onboarding.ts
import { apiRequest } from '../lib/apiClient'

export interface Plan {
  id: number
  name: string
  monthly_price_usd: string
  features?: Record<string, any> | null
  discounts?: Record<string, number> | null
}

export interface SignupPayload {
  organisation_name: string
  slug: string
  admin_email: string
  admin_phone?: string | null
  plan_id: number
  billing_months: number
}

export interface SignupResponse {
  id: string
  reference: string
  status: string
  plan_id: number
}

export interface CheckoutResponse {
  checkout_url: string
  transaction_id?: string | null
  status: string
}

export interface InvitationCheckResponse {
  organisation_id: number
  organisation_name: string
  slug: string
  plan_id: number
  plan_name: string
  monthly_price_usd: string
  discounts?: Record<string, number> | null
}

export async function listPlans(): Promise<Plan[]> {
  return apiRequest('GET', '/onboarding/plans')
}

export async function createSignup(payload: SignupPayload): Promise<SignupResponse> {
  return apiRequest('POST', '/onboarding/signup', payload)
}

export async function createCheckout(reference: string): Promise<CheckoutResponse> {
  return apiRequest('POST', '/onboarding/checkout', { reference })
}

export async function checkInvitation(email: string): Promise<InvitationCheckResponse> {
  return apiRequest('POST', '/onboarding/check-invitation', { email })
}

File: backend/app/schemas/onboarding.py
from __future__ import annotations

from pydantic import BaseModel, EmailStr, Field


class TenantSignupCreate(BaseModel):
    organisation_name: str = Field(min_length=2, max_length=200)
    slug: str = Field(min_length=2, max_length=100)
    admin_email: EmailStr
    admin_phone: str | None = None
    plan_id: int
    billing_months: int = Field(default=1)


class TenantSignupResponse(BaseModel):
    id: str
    reference: str
    status: str
    plan_id: int
    organisation_id: int | None = None


class TenantCheckoutRequest(BaseModel):
    reference: str
    success_url: str | None = None
    cancel_url: str | None = None


class InvitationCheckRequest(BaseModel):
    email: EmailStr


class InvitationCheckResponse(BaseModel):
    organisation_id: int
    organisation_name: str
    slug: str
    plan_id: int
    plan_name: str
    monthly_price_usd: str
    discounts: dict | None = None


class TenantCheckoutResponse(BaseModel):
    checkout_url: str
    transaction_id: str | None = None
    status: str


class PlanOut(BaseModel):
    id: int
    name: str
    monthly_price_usd: str
    features: dict | None = None
    discounts: dict | None = None
